src/libs/actions/Session/index.js

```diff
@@ -455,17 +455,18 @@ function signInWithValidateCode(accountID, code, twoFactorAuthCode, preferredLoc
         },
     ];
 
-    API.write(
-        'SigninUserWithLink',
-        {
-            accountID,
-            validateCode,
-            twoFactorAuthCode,
-            preferredLocale,
-            deviceInfo: getDeviceInfoForLogin(),
-        },
-        {optimisticData, successData, failureData},
-    );
+    const params = {
+        accountID,
+        validateCode,
+        preferredLocale,
+        deviceInfo: getDeviceInfoForLogin(),
+    };
+
+    // Pass twoFactorAuthCode to server only if it has a valid value, otherwise php might convert it to "null" as a string.
+    if (twoFactorAuthCode) {
+        params.twoFactorAuthCode = twoFactorAuthCode;
+    }
```

Comment:
Isn't this a sign that this is an internal bug? Let's add the filter logic in php, if the twoFactorAuthCode is empty/unset, ignore it.